website/content/api-docs/secret/openldap.mdx

```diff
@@ -216,36 +238,257 @@ $ curl \
+`.Username` field within the `creation_ldif` string. See appendix for available template functions.
+
+`default_ttl` `(string/int)` - Specifies the TTL for the leases associated with this role. Accepts time suffixed
+strings ("1h") or an integer number of seconds. Defaults to system/engine default TTL time.
+
+`max_ttl` `(string/int)` - Specifies the maximum TTL for the leases associated with this role. Accepts time suffixed
+strings ("1h") or an integer number of seconds. Defaults to system/mount default TTL time; this value is allowed to
+be less than the mount max TTL (or, if not set, the system max TTL), but it is not allowed to be longer.
+
+#### Sample Payload
+
+Sample LDIF files:
+
+`creation.ldif`:
+```
+dn: cn={{.Username}},ou=users,dc=learn,dc=example
+objectClass: person
+objectClass: top
+cn: learn
+sn: {{.Password | utf16le | base64}}
+memberOf: cn=dev,ou=groups,dc=learn,dc=example
+userPassword: {{.Password}}
+```
+
+`deletion.ldif` & `rollback.ldif`:
+
+```
+dn: cn={{.Username}},ou=users,dc=learn,dc=example
+changetype: delete
+```
+
+Full Payload:
+
+```json
+{
+  "creation_ldif": "...",
+  "deletion_ldif": "...",
+  "rollback_ldif": "...",
+  "username_template": "...",
+  "default_ttl": "1h",
+  "max_ttl": "24h"
+}
+```
+
+-> Note: The LDIF statements may optionally be base64 encoded. If they are base64 encoded when creating/updating the
+   role configuration, the decoded version will be returned from the `GET` endpoint.
+
+#### Sample POST Request
 
 ```shell-session
 $ curl \
     --header "X-Vault-Token: ..." \
     --request POST \
-    http://127.0.0.1:8200/v1/openldap/rotate-role/:role_name
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/openldap/role/dynamic-role
 ```
+
+### Read Dynamic Role Configuration
+
+| Method | Path                       |
+| ------ | -------------------------- |
+| `GET`  | `openldap/role/:role_name` |
+
+Retrieves a dynamic role's configuration.
+
+#### Parameters
+
+None
+
+#### Response
+
+`200 OK`
+
+```json
+{
+  "creation_ldif": "...",
+  "default_ttl": 1800,
+  "deletion_ldif": "...",
+  "max_ttl": 0,
+  "rollback_ldif": "...",
+  "username_template": "..."
+}
+```
+
+### Templates
+
+LDIF and username templates use the [Go template syntax](https://golang.org/pkg/text/template/) to construct the
+LDIF/username that will be executed against the server. This is done because certain values (such as the username
+and password within an LDIF statement) are not known at configuration time. Additionally, the template gives a
+lot of flexibility to the operator to construct the LDIF and username.
+
+Templated fields are delimited by `{{` and `}}`. To reference a field (such as the generated `Username`), a period is
+placed in front of the field name. Example: `{{.Username}}`. This operates very similarly to
+[`jq`](https://stedolan.github.io/jq/). Spaces can be included between `{{` and `}}`. For instance:
```

Comment:
Not sure about the `jq` reference, which might either: a) be unfamiliar  b) lead readers to think at other jq-ism will work here.

Reply:
That's a good point. I've removed the note about `jq`